cmd/restic/cmd_prune.go

```diff
@@ -187,40 +199,54 @@ type packInfoWithID struct {
 	packInfo
 }
 
-// prune selects which files to rewrite and then does that. The map usedBlobs is
-// modified in the process.
-func prune(opts PruneOptions, gopts GlobalOptions, repo restic.Repository, usedBlobs restic.BlobSet) error {
-	ctx := gopts.ctx
+type pruneStats struct {
+	Blobs struct {
+		Used         uint `json:"used"`
+		Duplicate    uint `json:"duplicate"`
+		Unused       uint `json:"unused"`
+		Total        uint `json:"total"`
+		Repack       uint `json:"repack"`
+		RepackRm     uint `json:"repack_remove"`
+		Remove       uint `json:"remove"`
+		RemoveTotal  uint `json:"remove_total"`
+		Remain       uint `json:"remaining"`
+		RemainUnused uint `json:"remaining_unused"`
+	} `json:"blobs"`
+	Size struct {
+		Used         uint64 `json:"used"`
+		Duplicate    uint64 `json:"duplicate"`
+		Unused       uint64 `json:"unused"`
+		Unref        uint64 `json:"unreferenced"`
+		Total        uint64 `json:"total"`
+		Repack       uint64 `json:"repack"`
+		Repackrm     uint64 `json:"repack_remove"`
+		Remove       uint64 `json:"remove"`
+		RemoveTotal  uint64 `json:"remove_total"`
+		Remain       uint64 `json:"remaining"`
+		RemainUnused uint64 `json:"remaining_unused"`
+	} `json:"bytes"`
+	Packs struct {
+		Used        uint `json:"used"`
+		Unused      uint `json:"unused"`
+		PartlyUsed  uint `json:"partly_used"`
+		Unref       uint `json:"unreferenced"`
+		Total       uint `json:"total"`
+		Keep        uint `json:"keep"`
+		Repack      uint `json:"repack"`
+		Remove      uint `json:"remove"`
+		RemoveTotal uint `json:"remove_total"`
+	} `json:"packfiles"`
+}
 
-	var stats struct {
-		blobs struct {
-			used      uint
-			duplicate uint
-			unused    uint
-			remove    uint
-			repack    uint
-			repackrm  uint
-		}
-		size struct {
-			used      uint64
-			duplicate uint64
-			unused    uint64
-			remove    uint64
-			repack    uint64
-			repackrm  uint64
-			unref     uint64
-		}
-		packs struct {
-			used       uint
-			unused     uint
-			partlyUsed uint
-			keep       uint
-		}
-	}
+// collectPrune selects which files to rewrite and which to delete and which blobs to keep.
+// Also some statistics are returned.
+// The map usedBlobs is modified in the process.
+func collectPrune(opts PruneOptions, gopts GlobalOptions, repo restic.Repository, usedBlobs restic.BlobSet) (removePacksFirst, repackPacks, removePacks restic.IDSet, keepBlobs restic.BlobSet, stats pruneStats, err error) {
```

Comment:
What about `planPrune` as method name?

Reply:
I like that name and also named the struct `prunePlan` :wink: